R/validate-pds-trips.R

```diff
@@ -0,0 +1,116 @@
+#' Validate Pelagic Data System trips
+#'
+#'Downloads the preprocessed version of pds trips from cloud storage services and
+#'validates a range of information so that it can be safely used for analysis.
+#'
+#'The parameters needed in the config file are those required for
+#'`preprocess_pds_trips()`, as well as parameters needed to identify anomalous
+#'trips, that are  `hrs`and `km`.
+#'
+#' @param log_threshold
+#' @inheritParams ingest_pds_trips
+#'
+#' @return no outputs. This function is used for it's side effects
+#' @export
+#' @importFrom rlang .data
+#'
+validate_pds_trips <- function(log_threshold = logger::DEBUG){
+
+  logger::log_threshold(log_threshold)
+  pars <- read_config()
+  pds_trips <- get_preprocessed_trips(pars)
+
+  hrs <- pars$validation$pds_trips$trip_hours
+  km <- pars$validation$pds_trips$trip_km
+
+  logger::log_info("Validating pds trips...")
+  navigation_alerts <- validate_pds_navigation(pds_trips,hrs=hrs, km=km)
+
+  # take ready (?) columns
+  ready_cols <- pds_trips %>%
+    dplyr::select(.data$`Last Seen`,
+                  .data$IMEI,
+                  .data$Trip,
+                  .data$Ended)
```

Comment:
I think we could also add the device ID here. Not sure if we'll use it in the future, but maybe doesn't hurt as there are times in which they can replace the sim-card (with different IMEI) into the same device.